l10n_ar_sale/models/sale_order.py

```diff
@@ -63,13 +63,20 @@ def _get_sale_checkbook(self):
     @api.model_create_multi
     def create(self, vals):
         for val in vals:
+            sale_checkbook_id = val.get('sale_checkbook_id')
+            sale_checkbook_type = val.get('type_id')
             if self.env.user.has_group('l10n_ar_sale.use_sale_checkbook') and \
                 val.get('name', _('New')) == _('New') and \
-                    val.get('sale_checkbook_id'):
-                sale_checkbook = self.env['sale.checkbook'].browse(
-                    val.get('sale_checkbook_id'))
+                sale_checkbook_id:
+                sale_checkbook = self.env['sale.checkbook'].browse(sale_checkbook_id)
```

Comment:
```suggestion
                sale_checkbook = self.env['sale.checkbook'].browse(
                    sale_checkbook_id)
```
es para que no haya cambios innecesarios en el diff